mbedtls/src/rng/rdrand.rs

```diff
@@ -7,33 +7,71 @@
  * according to those terms. */
 
 use core::slice::from_raw_parts_mut;
+
 use mbedtls_sys::types::raw_types::{c_int, c_uchar, c_void};
 use mbedtls_sys::types::size_t;
 
+// Intel documentation claims that if hardware is working RDRAND will produce output
+// after at most 10 attempts
+const RDRAND_READ_ATTEMPTS : i32 = 10;
+
+// Intel does not document the number of times RDSEED might consecutively fail, but in
+// example code uses 75 as the upper bound.
+const RDSEED_READ_ATTEMPTS : i32 = 75;
+
+fn call_cpu_rng<T: Sized + Copy>(attempts: i32, intrin: unsafe fn(&mut T) -> i32) -> Option<T> {
+    let mut attempts = attempts;
+    while attempts > 0 {
+        let mut out : T = unsafe { std::mem::uninitialized() };
+        let status = unsafe { intrin(&mut out) };
```

Comment:
Creating a mutable reference to uninitialized memory is UB. Initialize to `T::default()` instead

Reply:
til